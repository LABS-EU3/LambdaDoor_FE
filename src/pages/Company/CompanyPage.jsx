/* eslint-disable no-debugger */
/* eslint-disable no-shadow */
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { useParams } from 'react-router-dom';
import { Tabs, Button, Icon } from 'antd';
import { getCompanies } from '../../state/actions/companies';
import {
  getInterviewReviewsByCompanyId,
  getSalaryReviewsByCompanyId,
} from '../../state/actions/reviews';
import CompanyCard from '../../components/Company/CompanyDetails/CompanyCard';
import CompanyReviewCard from '../../components/Company/CompanyReview/CompanyReviewCard';
import InterviewReviewList from '../../components/Company/InterviewReviews/InterviewReviewList';
import SalaryReviewsList from '../../components/Company/SalaryReviews/SalaryReviewsList';

const { TabPane } = Tabs;
const CompanyPage = ({
  getCompanies,
  getInterviewReviewsByCompanyId,
  companies: { companies },
  // avgSalaries: { avgSalaries },
  authState: {
    credentials: { id },
  },
  getSalaryReviewsByCompanyId,
  history,
}) => {
  const companyId = useParams().id;

  useEffect(() => {
    getCompanies();
    getInterviewReviewsByCompanyId(companyId);
    getSalaryReviewsByCompanyId(companyId);
  }, []);

  return (
    <div>
      <Button
        style={{
          marginBottom: '30px',
          border: '1px solid #BB1333',
          color: '#BB1333',
        }}
        onClick={() => history.goBack()}
      >
        <Icon type="left" />
        Back
      </Button>
      <Tabs defaultActiveKey="1">
        <TabPane tab="Company Info" key="1">
          <CompanyCard companies={companies} />
          {/* <CompanySalaryChart avgSalaries={avgSalaries} /> */}
        </TabPane>
        <TabPane tab="Company Reviews" key="2">
          <CompanyReviewCard />
        </TabPane>
        <TabPane tab="Salary Reviews" key="3">
          <SalaryReviewsList />
        </TabPane>
        <TabPane tab="Interview Process Reviews" key="4">
          <InterviewReviewList />
        </TabPane>
      </Tabs>
    </div>
  );
};
export default connect(state => state, {
  getCompanies,
  getInterviewReviewsByCompanyId,
  getSalaryReviewsByCompanyId,
})(CompanyPage);
