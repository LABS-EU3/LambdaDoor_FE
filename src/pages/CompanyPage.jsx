/* eslint-disable no-debugger */
/* eslint-disable no-shadow */
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Tabs } from 'antd';
import { getCompanies } from '../state/actions/companies';
import CompanyCard from '../components/CompanyCard/CompanyCard';

const { TabPane } = Tabs;

const CompanyPage = ({
  getCompanies,
  companies: { companies },
  authState: {
    credentials: { id },
  },
}) => {
  useEffect(() => {
    getCompanies();
  }, []);

  return (
    <div>
      <Tabs defaultActiveKey="1">
        <TabPane tab="Company Info" key="1">
          <CompanyCard companies={companies} />
        </TabPane>
        <TabPane tab="Company Reviews" key="2">
          Content of Tab Pane 2
        </TabPane>
        <TabPane tab="Salary Reviews" key="3">
          Content of Tab Pane 3
        </TabPane>
        <TabPane tab="Interview Process Reviews" key="4">
          Content of Tab Pane 4
        </TabPane>
      </Tabs>
    </div>
  );
};

export default connect(state => state, { getCompanies })(CompanyPage);
