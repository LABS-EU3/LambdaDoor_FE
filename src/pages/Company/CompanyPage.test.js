import React from 'react';
import * as rtl from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import CompanyPage from './CompanyPage';

// const historyMock = { push: jest.fn() };

jest.mock('./CompanyPage', () => () => (
  <div id="mockCompanyPage">mockCompanyPage</div>
));

const testCompanies = [
  {
    id: 5,
    name: 'DoNotPay Inc',
    description:
      'Fight corporations, beat bureaucracy and sue anyone at the press of a button.',
    website: 'https://donotpay.com',
    location: 'Portland, OR',
    type: 'Technology',
    logo: '',
    latitude: 31,
    longitude: -80,
    average_rating: '5.0000000000000000',
  },
  {
    id: 4,
    name: 'Bad Rabbit',
    description:
      'At Bad Rabbit, we make the systems you have work better for you.',
    website: 'https://www.badrabbit.com',
    location: 'Portland, OR',
    type: 'Technology',
    logo: '',
    latitude: 31,
    longitude: -80,
    average_rating: '5.0000000000000000',
  },
  {
    id: 10,
    name: 'Paystack',
    description:
      'Paystack helps businesses in Africa get paid by anyone, anywhere in the world',
    website: 'https://paystack.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    latitude: 31,
    longitude: -80,
    average_rating: '4.0000000000000000',
  },
];

const testSalaries = {
  avgSalaries: [
    {
      interest_id: 2,
      interest: 'Front End',
      currency: 'USD',
      avg: 95000,
    },
    {
      interest_id: 4,
      interest: 'Full Stack',
      currency: 'USD',
      avg: 67500,
    },
  ],
};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    id: '5',
  }),
}));

afterEach(cleanup);

describe('CompanyPage', () => {
  it('renders correctly', () => {
    expect(
      rtl.render(
        <CompanyPage companies={testCompanies} avgSalaries={testSalaries} />
      ).baseElement
    ).toMatchSnapshot();
  });
});
