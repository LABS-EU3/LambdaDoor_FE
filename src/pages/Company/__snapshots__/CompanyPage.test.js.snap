// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyPage renders correctly 1`] = `
<body>
  <div>
    <div
      id="mockCompanyPage"
    >
      mockCompanyPage
    </div>
  </div>
</body>
`;
